charts/vector/values.yaml

```diff
@@ -10,23 +10,46 @@
 ## Stateless-Aggregator - Deployment
 role: "Aggregator"
 
+## Define the Vector image to use
 image:
+  # image.repository -- Override default registry + name for Vector
   repository: timberio/vector
+  # image.pullPolicy -- Vector image pullPolicy
   pullPolicy: IfNotPresent
-  # Overrides the image tag whose default is the chart appVersion
+  # image.pullSecrets -- Agent repository pullSecret (ex: specify docker registry credentials)
+  ## Ref: https://kubernetes.io/docs/concepts/containers/images/#specifying-imagepullsecrets-on-a-pod
+  pullSecrets: []
+  # image.tag -- Vector image tag to use
+  # @default -- Chart's appVersion
   tag: ""
 
-# replicas -- Set the number of pods to create
+# replicas -- Set the number of Pods to create
 ## Valid for Aggregator and Stateless-Aggregator
 replicas: 1
 
+## Create a Secret resource for Vector to use
+secrets:
+  # secrets.generic -- Each Key/Value will be added to the Secret's data key
+  ## Values should be entered base64 encoded (examples below are "REPLACE_ME" encoded)
+  ## NOTE: Don't commit unencrypted secrets to git!
+  generic: {}
+    # awsAccessKeyId: "UkVQTEFDRV9NRQo="
+    # awsSecretAccessKey: "UkVQTEFDRV9NRQo="
+
+## Configure a HorizontalPodAutoscaler for Vector
+## Valid for Stateless-Aggregator role
 autoscaling:
   # autoscaling.enabled -- Enabled autoscaling for the Stateless-Aggregator
   enabled: false
+  # autoscaling.minReplicas -- Minimum replicas for Vector's HPA
   minReplicas: 1
+  # autoscaling.maxReplicas -- Maximum replicas for Vector's HPA
   maxReplicas: 10
+  # autoscaling.targetCPUUtilizationPercentage -- Target CPU utilization for Vector's HPA
   targetCPUUtilizationPercentage: 80
-  # targetMemoryUtilizationPercentage: 80
+  # autoscaling.targetMemoryUtilizationPercentage -- (int) Target memory utilization for Vector's HPA
+  targetMemoryUtilizationPercentage:
```

Comment:
Does this just set the value to `null` by default?

Reply:
The template doesn't render unless there's a `value` so by default no utilization target for memory, here: https://github.com/timberio/helm-charts/blob/master/charts/vector/templates/hpa.yaml#L24

Reply:
Aha 👍